flink-staging/flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala

```diff
@@ -61,99 +73,217 @@ object FlinkShell {
+    val jarPath = new Path("file://" +
+      s"${yarnClient.getClass.getProtectionDomain.getCodeSource.getLocation.getPath}")
+    yarnClient.setLocalJarPath(jarPath)
+
+    // load configuration
+    val confDirPath = CliFrontend.getConfigurationDirectoryFromEnv
+    val flinkConfiguration = GlobalConfiguration.getConfiguration
+    val confFile = new File(confDirPath + File.separator + "flink-conf.yaml")
+    val confPath = new Path(confFile.getAbsolutePath)
+    GlobalConfiguration.loadConfiguration(confDirPath)
+    yarnClient.setFlinkConfigurationObject(flinkConfiguration)
+    yarnClient.setConfigurationDirectory(confDirPath)
+    yarnClient.setConfigurationFilePath(confPath)
+
+    // number of task managers is required.
+    yarnConfig.containers match {
+      case Some(containers) => yarnClient.setTaskManagerCount(containers)
+      case None =>
+        throw new IllegalArgumentException("Number of taskmanagers must be specified.")
+    }
+
+    // set configuration from user input
+    yarnConfig.jobManagerMemory.foreach(yarnClient.setJobManagerMemory)
+    yarnConfig.name.foreach(yarnClient.setName)
+    yarnConfig.queue.foreach(yarnClient.setQueue)
+    yarnConfig.slots.foreach(yarnClient.setTaskManagerSlots)
+    yarnConfig.taskManagerMemory.foreach(yarnClient.setTaskManagerMemory)
+
+    // deploy
+    val cluster = yarnClient.deploy()
+    val address = cluster.getJobManagerAddress.getAddress.getHostAddress
+    val port = cluster.getJobManagerAddress.getPort
+    cluster.connectToCluster()
+
+    (address, port, Some(Right(cluster)))
+  }
+
+  def fetchDeployedYarnClusterInfo() = {
+    // load configuration
+    val globalConfig = GlobalConfiguration.getConfiguration
+    val defaultPropertiesLocation = System.getProperty("java.io.tmpdir")
+    val currentUser = System.getProperty("user.name")
+    val propertiesLocation = globalConfig.getString(
+      ConfigConstants.YARN_PROPERTIES_FILE_LOCATION, defaultPropertiesLocation)
+    val propertiesName = CliFrontend.YARN_PROPERTIES_FILE + currentUser
+    val propertiesFile = new File(propertiesLocation, propertiesName)
+
+    // read properties
+    val properties = if (propertiesFile.exists()) {
+      println("Found YARN properties file " + propertiesFile.getAbsolutePath)
+      val properties = new Properties()
+      val inputStream = new FileInputStream(propertiesFile)
+
+      try {
+        properties.load(inputStream)
+      } finally {
+        inputStream.close()
+      }
+
+      Some(properties)
+    } else {
+      None
     }
 
-    System.out.println(" good bye ..")
+    properties match {
+      case Some(props) =>
+        val addressInStr = props.getProperty(CliFrontend.YARN_PROPERTIES_JOBMANAGER_KEY)
+        val address = ClientUtils.parseHostPortAddress(addressInStr)
+
+        (address.getHostString, address.getPort, None)
+      case None =>
+        throw new IllegalArgumentException("Scala Shell cannot fetch YARN properties.")
```

Comment:
Shouldn't we fail in the preceding `else` branch? Then we could also move the `Some` case in the `if` branch.